### DISCUSSION TITLE
Show distance to next POI

### DISCUSSION THREAD

**Discussion by randomizeduser** [organicmaps/organicmaps#2540] [Ideas]
Hello, I love your app!

Could you include showing distance to the next POI around you while navigating?

Here is the scenario:

- You are using the navigation (in a car or motorcycle) and following the route.
- While driving you cannot take your phone and start exploring the map for important POIs like gas stations, parking etc. because you drive alone.
- However you want to know the distance to the next POI (gas station) at a quick glance, to know whether you can continue driving or need to get gas now. POIs to be shown during navigation should be user configurable.

Note, with POIs I do not mean stops/waypoints which you configured to be on the route during navigation. But POIs which are simply around you that you are passing while navigating.

For me personally this would be great, because I am riding a motorcycle and my gas tank is very small. So I always want to know the distance to the next gas station.

Other applications would be:
- You are driving on the highway and need to use the bathroom: how far is the next rest/stop place?
- You are hiking and are thirsty: how far is the next water source?
etc.

Here is a screenshot how it is implemented in Sygic: https://imgur.com/a/0uVNTKG

In Sygic when you tap on the POI icon it asks you if you want to add it as a waypoint. I think it would be also a great addition, but that's probably too much work.

**Comment by biodranik** [comment]
That's a great idea, let's create an issue for it. It would be great to minimize the configuration, ideally to avoid it at all )

**Reply by randomizeduser** [reply]
Awesome! I'm glad you like it! I have created issue https://github.com/organicmaps/organicmaps/issues/2545 for this with some mock-ups. Let me know if that suffices.
